### PR TITLE
Add a notification script if monit restarts chia

### PR DESCRIPTION
**Pull request by pmaslana**

### CODE REVIEW

File: chia-blockchain/defaults/main.yml [hunk 1]
```diff
@@ -260,6 +260,7 @@ chia_add_healthcheck_service: true
 chia_healthcheck_port: 9950
 add_monit_config: true
 monit_start_now: true
+chia_alerts_receiver_token: "{{ lookup('env', 'CHIA_ALERTS_RECEIVER_TOKEN') }}"
```
Comment (cmmarslender) [inline chia-blockchain/defaults/main.yml:263]: For a defaults file, this should just be an empty string. The env lookup should be in the playbook/vars that actually call this role

File: chia-blockchain/templates/monit_config.j2 [hunk 2]
```diff
@@ -10,3 +10,4 @@ check host localhost with address 127.0.0.1
     {% endif %}
 {% endif %}
     if 5 restarts within {{ 5 * chia_monit_failure_threshold }} cycles then timeout
+    if 1 restarts within {{ chia_monit_failure_threshold }} cycles then exec "/etc/monit/monit-notify.sh"
```
Comment (cmmarslender) [inline chia-blockchain/templates/monit_config.j2:13]: We should probably wrap this with an option to enable it with it disabled/false in defaults in the role itself

File: chia-blockchain/templates/monit_notify.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Webhook URL where the alert will be sent
+WEBHOOK_URL="https://alert-receiver.chiaops.com/ips-info"
```
Comment (cmmarslender) [inline chia-blockchain/templates/monit_notify.sh.j2:4]: Since these are open source roles, this should be a var

**Review by cmmarslender — CHANGES_REQUESTED** [review]

File: chia-blockchain/templates/monit_notify.sh.j2 [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+HOSTNAME=$(/bin/hostname)
+BEARER_TOKEN="{{ CHIA_ALERTS_RECEIVER_TOKEN | quote }}"
+
+# Get the Tailscale IPv4 address
+TAILSCALE_IPV4=$(tailscale ip | grep -oE '([0-9]{1,3}\.){3}[0-9]{1,3}' | head -n 1)
```
Comment (cmmarslender) [inline chia-blockchain/templates/monit_notify.sh.j2:9]: We have tailscale IPs internally, but not always externally. I think we can omit this part, and just send the hostname in the alert

Comment (pmaslana) [reply chia-blockchain/templates/monit_notify.sh.j2:9]: Removed tailscale IPs

File: chia-blockchain/templates/monit_config.j2 (hunk 2 above)
Comment (pmaslana) [reply chia-blockchain/templates/monit_config.j2:13]: This has been wrapped.

File: chia-blockchain/templates/monit_notify.sh.j2 (hunk 3 above)
Comment (pmaslana) [reply chia-blockchain/templates/monit_notify.sh.j2:4]: Switched to using a var

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**